src/Middleware/HttpLogging/src/HttpLoggingContext.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.HttpLogging;
+
+/// <summary>
+/// The context used for logging customization callbacks.
+/// </summary>
+/// <param name="httpContext">The request context.</param>
+public sealed class HttpLoggingContext(HttpContext httpContext)
+{
+    /// <summary>
+    /// The request context.
+    /// </summary>
+    public HttpContext HttpContext { get; internal set; } = httpContext;
+
+    /// <summary>
+    /// What parts of the request and response to log.
+    /// </summary>
+    public HttpLoggingFields LoggingFields { get; set; }
+
+    /// <summary>
+    /// Limits how much of the request body to log.
+    /// </summary>
+    public int RequestBodyLogLimit { get; set; }
+
+    /// <summary>
+    /// Limits how much of the response body to log.
+    /// </summary>
+    public int ResponseBodyLogLimit { get; set; }
+
+    internal long StartTimestamp { get; set; }
+
+    /// <summary>
+    /// The parameters to log.
+    /// </summary>
+    public IList<KeyValuePair<string, object?>> Parameters { get; } = new List<KeyValuePair<string, object?>>();
+
+    /// <summary>
+    /// Adds a parameter to the log context.
+    /// </summary>
+    /// <param name="key">The parameter name.</param>
+    /// <param name="value">The parameter value.</param>
+    public void Add(string key, object? value)
```

Comment:
AddParameter

But why not make `Parameters` collection public?

Reply:
It is public now. This method is mostly for convenience, hence trying to keep the name short.